### ISSUE TITLE
Changing training config file through docker 

### ISSUE THREAD

**Issue by Pri0805** [MIC-DKFZ/nnDetection#275]
Hi, 

We were able to successfully run train the model on our dataset but we noticed poor validation loss curves compared to the training loss. We noticed that the config file already considers class imbalance and uses oversample_foreground_percent = 0.5 to sample half of the positive / negative cases. 

![image](https://github.com/user-attachments/assets/a84fcf4f-1c57-40e6-ba3f-ef307ad4d07f)
![image](https://github.com/user-attachments/assets/1a111cde-c8c8-46b8-a5bb-54b91ed5fded)

We are trying to change other parameters in the training configuration in the v001.yaml file and are using docker to run the commands. Is there a way to update the training parameters (warm iterations, initial_lr, warm_lr, etc.) through docker?

Thanks!

**Comment by partha-ghosh** [comment]
Hi @Pri0805,
You can update the parameters through the command line itself. For example, to update the `warm_iterations` to 5000 and `initial_lr` to 0.001, you have to pass `-o trainer_cfg.warm_iterations=5000 trainer_cfg.initial_lr=0.001`. This way you can update other parameters as well.

Best,
Partha

**Comment by Pri0805** [comment]
Got it! Thank you

**Comment by mibaumgartner** [comment]
Dear @Pri0805 ,

please note that the training/validation losses might not be the best indicators due to how the detection losses are computed (i.e. Hard Negative Mining is a stochastic process which will include the worst scoring anchors which might show different characteristics as the detection performance). I would generally recommend relying on the validation metrics for analysis ;)

**Comment by Pri0805** [comment]
Oh I see! This is how our mAP curve for one of the folds looks like. Our consolidated score was also quite low around 0.427. We are running the model on a tumor dataset with 60% negative cases and 40% positive cases, is there any other reason why we could be getting poor scores?

<img width="452" alt="Screenshot 2024-10-08 at 7 29 39 PM" src="https://github.com/user-attachments/assets/2bd857da-5d00-4728-bf41-14d4e1bdfebe">

**Comment by mibaumgartner** [comment]
Hey @Pri0805,

1) you can check #https://github.com/MIC-DKFZ/nnDetection/issues/272 and the Metrics Reloaded & Metric Pitfall papers (from the Lena Maier-Hein Group) for intuition on the metrics. AP is generally somewhat pessimistic in datasets with many negative cases (IMO). 

2) prostate cancer is a very difficult detection problem in general, so I wouldn't expect super high performance 

3) prostate lesions have quite a large amount of ambiguity in their borders during the annotation process, so you should prefer to look at low IoU values like 0.1 

Best,
Michael

**Comment by Pri0805** [comment]
Sure noted, thank you for the information!
